src/main/java/seedu/mark/model/reminder/Reminder.java

```diff
@@ -0,0 +1,114 @@
+package seedu.mark.model.reminder;
+
+import static seedu.mark.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import seedu.mark.model.bookmark.Bookmark;
+
+
+/**
+ * Represents a reminder in Mark.
+ */
+public class Reminder {
+
+    //TODO: temporarily; to change when confirmed
+    private static final String DATE_FORMATTER = "dd/MM/yyyy HHmm";
+
+    //data fields
+    private Bookmark bookmark;
+    private LocalDateTime remindTime;
+
+    //identity field
+    private Note note;
+
+    /**
+     * Every field must be present and not null.
+     *
+     * @param bookmark the bookmark the reminder should open.
+     * @param time the reminding time.
+     * @param note the note for reminder.
+     */
+    public Reminder(Bookmark bookmark, LocalDateTime time, Note note) {
+        requireAllNonNull(bookmark, time, note);
+        this.bookmark = bookmark;
+        this.remindTime = time;
+        this.note = note;
+    }
+
+    public Bookmark getBookmark() {
+        return bookmark;
+    }
+
+    public LocalDateTime getRemindTime() {
+        return remindTime;
+    }
+
+    public Note getNote() {
+        return note;
+    }
+
+    /**
+     * Returns true if both reminders have the same note, the same bookmark and the same time.
+     */
```

Comment:
```suggestion
     * TODO: change implementation based on how to distinguish reminders (which field to use)
     */
```
_As recommended by @Na-Nazhou _